### DISCUSSION TITLE
Prevent the usage of SCIP or GLPK when using pre-built binary

### DISCUSSION THREAD

**Discussion by hedtke** [google/or-tools#3125] [General] [answered]
When using a pre-built binary (e.g., as described in the installation instructions for Java on Debian) from https://github.com/google/or-tools/releases/tag/v9.2 I guess there is GLPK and SCIP included.

Is there a way to forbid the usage of SCIP and GLPK during runtime, or is a build from source required in this case?

Would it work to simply set the solver to CBC for example, via `MPSolver::CreateSolver` or is it possible that CP-SAT or existing VRP algorithms, ..., use SCIP in the background and setting the solver to CBC would have no effect?

**Comment by lperron** [comment, marked as answer]
GLPK is not included, for licensing problems.

I will not do two builds, one with scip, one without.

SCIP is a much more reliable solver than CBC :-( And it is one line to switch to a different solver.

What is the issue ?

You can build from source without SCIP. This is done by brew, freebsd. You need to pass a USE_SCIP=OFF to cmake or make.

**Reply by hedtke** [reply]
Thanks for the fast response. We know the USE_SCIP flag in cmake and use it when we build the OR Tools from scratch.
The idea behind my question is the simplification and speed-up of our CI and deployment pipelines.
As we do not have a SCIP license, we need to deactivate it. In order to do so, we need a custom built and cannot use the pre-built binaries. Thus we either build the OR Tools within the pipeline itself (which takes some seconds, requires a C++ compiler even for python or Java projects, ...), or we create a custom container that includes a version without SCIP (which we have to update with every new release). We are fine to continue with that process.
We were just hoping that there is a parameter that can be set at runtime even when USE_SCIP=ON was used during build time.

**Reply by gregy4** [reply]
For me both CBC and SCIP don't work, both mostly return no feasible solution for me.
I'm not sure whether SCIP solver is used by users of or-tools since people in academic world can use or-tools with commercial solvers like cplex or gurobi.
But SCIP license should be changed in next few weeks so we will see.

**Comment by lperron** [comment]
You can use the CP-SAT freely.
It does not use SCIP.

You cannot use SCIP directly (this is the only way you can access it) in a
commercial setting without the proper license from ZIB.
